src/core/sorted_map.cc

```diff
@@ -14,14 +14,24 @@ extern "C" {
 }
 
 #include "base/endian.h"
+#include "base/flags.h"
 #include "base/logging.h"
 
 using namespace std;
+
+ABSL_FLAG(bool, use_zset_tree, false, "If true use b+tree for zset implementation");
+
 namespace dfly {
 namespace detail {
 
 namespace {
 
+// We tag sds pointers to allow customizable comparison function that supports both
+// Lex and Numeric comparison.
+constexpr uint64_t kInfTag = 1ULL << 63;
+constexpr uint64_t kIgnoreDoubleTag = 1ULL << 62;
+constexpr uint64_t kSdsMask = (1ULL << 60) - 1;
+
```

Comment:
That looks potentially very dangerous, but I guess there is no space efficient alternative. Somewhen in the future I'd introduce proper wrapping types for all the sds/double stuff and BuilScoredKey would return some kind of proxy 👀

Reply:
added a comment